components/catalogs/back/tests/Integration/Infrastructure/Validation/ProductSelection/SystemCriterion/FamilyCriterionTest.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Catalogs\Test\Integration\Infrastructure\Validation\ProductSelection\SystemCriterion;
+
+use Akeneo\Catalogs\Infrastructure\Validation\ProductSelection\SystemCriterion\FamilyCriterion;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * @covers \Akeneo\Catalogs\Infrastructure\Validation\ProductSelection\SystemCriterion\FamilyCriterion
+ */
+class FamilyCriterionTest extends AbstractSystemCriterionTest
+{
+    private ?ValidatorInterface $validator;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->validator = self::getContainer()->get(ValidatorInterface::class);
+
+        $this->createFamily(['code' => 'familyA', 'label' => '[familyA]']);
+        $this->createFamily(['code' => 'familyB', 'label' => '[familyB]']);
+    }
+
+    /**
+     * @dataProvider validDataProvider
+     */
+    public function testItReturnsNoViolation(array $criterion): void
+    {
+        $violations = $this->validator->validate($criterion, new FamilyCriterion());
+
+        $this->assertEmpty($violations);
+    }
+
+    public function validDataProvider(): array
+    {
+        return [
+            'field with EMPTY operator' => [
+                'criterion' =>                     [
```

Comment:
```suggestion
        return [
            'field with EMPTY operator' => [
                'criterion' => [
```
and so on for next lines